docs/readmes/README.role-assignment.md

```diff
@@ -0,0 +1,47 @@
+2. [Managed identity](https://docs.microsoft.com/en-us/azure/aks/use-managed-identity) (it can be using a system-assigned identity or user-assigned identity)
+
+> Clusters with managed identity are only compatible with AAD Pod Identity 1.5+.
+
+## Performing Role Assignment
+
+After your cluster is provisioned, depending on your clsuter configuration, run one of the following commands to retrieve the **Principal ID** of your identity:
+
+| Cluster configurations                           | Commands                                                                                                 |
+|--------------------------------------------------|----------------------------------------------------------------------------------------------------------|
+| AKS cluster with service principal               | `az aks show -g <ResourceGroup> -n <ClusterName> --query servicePrincipalProfile.clientId -otsv`          |
+| AKS cluster with managed identity                | `az aks show -g <ResourceGroup> -n <ClusterName> --query identityProfile.kubeletidentity.clientId -otsv` |
+| aks-engine cluster with service principal        | Use the client ID of the service principal defined in the API model                                      |
+| aks-engine cluster with system-assigned identity | `az <vm\|vmss> identity show -g <ResourceGroup> -n <VM\|VMSS Name> --query principalId -otsv`            |
+| aks-engine cluster with user-assigned identity   | `az <vm\|vmss> identity show -g <ResourceGroup> -n <VM\|VMSS Name> --query userAssignedIdentities -otsv` |
+
+The roles [Managed Identity Operator](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#managed-identity-operator) and [Virtual Machine Contributor](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#virtual-machine-contributor) must be assigned to the service principal / identiy before deploying AAD Pod Identity so that it can perform identity-related operations on Azure. You can run the following command to assign the role with the resource group scope:
+
+```bash
+az role assignment create --role "Virtual Machine Contributor" --assignee <PrincipalID> --scope /subscriptions/<SubscriptionID>/resourcegroups/<ResourceGroup>
+az role assignment create --role "Managed Identity Operator" --assignee <PrincipalID> --scope /subscriptions/<SubscriptionID>/resourcegroups/<ResourceGroup>
+```
+
+To enable fine-grained control on which user-assigned identity the cluster has access to, run the following command:
+
+```bash
+az role assignment create --role "Managed Identity Operator" --assignee <PrincipalID>  --scope /subscriptions/<SubscriptionID>/resourcegroups/<ResourceGroup>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<IdentityName>
+```
+
+## VNet outside of the cluster resource group
```

Comment:
I don't think we need this anymore now that we are doing `PATCH` instead of `CreateOrUpdate` on the VM/VMSS. We need to validate this.